tgui/packages/tgui/interfaces/MusicInstrument.tsx

```diff
@@ -0,0 +1,183 @@
+import { useBackend, useLocalState } from '../backend';
+import { Button, Flex, LabeledList, Box, Knob, Input } from '../components';
+import { Window } from '../layouts';
+
+type MusicInstrumentData = {
+  name: string;
+  notes: string[];
+};
+
+export const MusicInstrument = (props, context) => {
+  const { act, data } = useBackend<MusicInstrumentData>(context);
+  const { name, notes } = data;
+
+  const [noteKeysOrder, setNoteKeysOrder] = useLocalState(
+    context,
+    'keyboardBindingsOrder',
+    'zsxdcvgbhnjmq2w3er5t6y7ui9o0p+'.split('')
+  );
+
+  const [activeKeys, setActiveKeys] = useLocalState(context, 'keyboardActivekeys', new Array(notes.length));
+  const [keyOffset, setKeyOffset] = useLocalState(context, 'keyOffset', 0);
+  const [keybindToggle, setKeybindToggle] = useLocalState(context, 'keybindToggle', false);
+  const [volume, setVolume] = useLocalState(context, 'keyboardVolume', 50);
+  const [transpose, setTranspose] = useLocalState(context, 'keyboardTranspose', 0);
```

Comment:
I think these two at least would be something you'd want to persist if you closed the UI and reopened, so should be part of `data` rather than local state?

Reply:
That makes sense, I'll change it

Reply:
added in fc8c7171d39c322b8c3b1922bfe1d818e0234789